tasks/main.yml

```diff
@@ -1,2 +1,51 @@
 ---
-# tasks file for skeleton
+- name: Verify that at least one Transparent HugePage setting is provided
+  ansible.builtin.fail:
+    msg: At least one Transparent HugePage setting must be provided
+  when:
+    - manage_thp_defrag_setting is not defined
+    - manage_thp_enabled_setting is not defined
+    - manage_thp_shmem_enabled_setting is not defined
+    - manage_thp_use_zero_page_setting is not defined
+
+- name: Verify the provided Transparent HugePage "defrag" setting
+  ansible.builtin.assert:
+    fail_msg: The provided Transparent HugePage "defrag" setting is not valid
+    that:
+      - manage_thp_defrag_setting in ['always', 'defer', 'defer+madvise', 'madvise', 'never']
+  when:
+    - manage_thp_defrag_setting is defined
+
+- name: Verify the provided Transparent HugePage "enabled" setting
+  ansible.builtin.assert:
+    fail_msg: The provided Transparent HugePage "enabled" setting is not valid
+    that:
+      - manage_thp_enabled_setting in ['always', 'madvise', 'never']
+
+- name: Verify the provided Transparent HugePage shmem_enabled setting
+  ansible.builtin.assert:
+    fail_msg: The provided Transparent HugePage shmem_enabled setting is not valid
+    that:
+      - manage_thp_shmem_enabled_setting in ['always', 'within_size', 'advise', 'never', 'deny', 'force']
```

Comment:
Alphabetize?
```suggestion
      - manage_thp_shmem_enabled_setting in ['advise', 'always', 'deny', 'force', 'never', 'within_size']
```

Reply:
If @mcdonnnj is copying from the Linux documentation then I recommend retaining the same order used there.

Reply:
Resolving this conversation - please see https://github.com/cisagov/ansible-role-manage-thp/pull/4#discussion_r1578078546 for details.